opm/input/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp

```diff
@@ -1920,28 +1916,18 @@ size_t SummaryConfig::size() const {
     return this->m_keywords.size();
 }
 
-/*
-  Can be used to query if a certain 3D field, e.g. PRESSURE, is
-  required to calculate the summary variables.
-
-  The implementation is based on the hardcoded datastructure
-  required_fields defined in a anonymous namespaces at the top of this
-  file; the content of this datastructure again is based on the
-  implementation of the Summary calculations in the opm-output
-  repository: opm/output/eclipse/Summary.cpp.
-*/
-
-bool SummaryConfig::require3DField( const std::string& keyword ) const {
-    const auto iter = required_fields.find( keyword );
-    if (iter == required_fields.end())
+// Can be used to query if a certain 3D field, e.g. PRESSURE, is required to
+// calculate a summary variable.
+bool SummaryConfig::require3DField(const std::string& keyword) const
+{
+    auto iter = required_fields.find(keyword);
+    if (iter == required_fields.end()) {
         return false;
-
-    for (const auto& kw : iter->second) {
-        if (this->hasKeyword( kw ))
-            return true;
     }
 
-    return false;
+    return std::any_of(iter->second.begin(), iter->second.end(),
+                       [this](const std::string& smryKw)
+                       { return this->match(smryKw); });
```

Comment:
this isn't entirely equivalent, but maybe that's on purpose (hasKeyword only checks short_keywords keys, while match does a shmatch on it.).

Reply:
Nm, I see it's in the PR description.

Reply:
>> this isn't entirely equivalent, but maybe that's on purpose (`hasKeyword` only checks short_keywords keys, while match does a `shmatch` on it.).
>
> Nm, I see it's in the PR description.

Yeah, it's an intentional change to the semantics as it enables matching region-level keywords which are not tied to the default `FIPNUM` region set.  Sorry for not highlighting this more clearly.